packages/relay/src/lib/eth.ts

```diff
@@ -634,7 +641,7 @@ export class EthImpl implements Eth {
     this.logger.trace(`${requestIdPrefix} getBlockByHash(hash=${hash}, showDetails=${showDetails})`);
     return this.getBlock(hash, showDetails, requestId).catch((e: any) => {
       this.logger.error(e, `${requestIdPrefix} Failed to retrieve block for hash ${hash}`);
-      return null;
+      throw predefined.INTERNAL_ERROR(e.message);
```

Comment:
Q: Do these new internal errors leak SDK or network information?
If so we should be putting static string here since we have the logging

Reply:
It will probably forward the error message if the error is MirrorNodeError or SDKError, but I don't know what info the messages contain. If it's a problem I can revert it. Or we could send the requestId and have something like `Internal error for request ${requestId}`. I was thinking that having more info about the error would be better to then locate it in the logs, when/if someone reports an issue, especially for the officially hosted relay, where dapp devs don't have access to the logs themselves.

Reply:
Yeah just log for now, we wouldn't want dependency (mirror node or sdk) related info showing up in the relay errors.
You can put static string in the errors to inform clients on actionable steps but if not leave it empty.
You can leave the logic to support a message in there as it's useful

e.g.
```
throw predefined.INTERNAL_ERROR('Downstream issue retrieving block information from network nodes');
```